src/settings_internal.h

```diff
@@ -150,6 +144,13 @@ struct IntSettingDesc : SettingDesc {
 	SettingCategory cat;    ///< assigned categories of the setting
 	OnChange *proc;         ///< callback procedure for when the value is changed
 
+	/**
+	 * Check whether this setting is an integer type setting.
+	 * @return True when the underlying type is an integer.
+	 */
+	virtual bool IsBoolSetting() const { return false; }
```

Comment:
this feels a bit weird, to be needing this. Nothing we can do about it?

PS: doxygen comment is a bit off now here :)